packages/wdio-logger/src/node.ts

```diff
@@ -70,7 +70,7 @@ const wdioLoggerMethodFactory = function (methodName, logLevel, loggerName) {
          */
         const match = Object.values(matches).filter(x => args[0].endsWith(`: ${x}`))[0]
         if (match) {
-            const prefixStr = args.shift().slice(0, -match.length - 1)
+            const prefixStr = args.shift()!.slice(0, -match.length - 1)
```

Comment:
how can we be sure `args.shift()` is not undefined?

Reply:
because `logLevel` always has at least one argument containing the date (even if you do `log.info()`)

Reply:
do you think it worst adding `args.lengths > 0` to line 72?

Reply:
makes sense!

Reply:
🤔 mhm even after changing it to:

```ts
        if (match && args[0]) {
            const prefixStr = args.shift().slice(0, -match.length - 1)
            args.unshift(prefixStr, match)
        }
```

it still fails.

Reply:
sorry for confusing, didn't expect it's not that smart, let's leave it with `!` then
```
if (match) {
            const prefixStr = args.shift()!.slice(0, -match.length - 1)
```